### DISCUSSION TITLE
Added: C# Language Design Notes for Mar 28, 2017

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#642] [General]
[C# Language Design Notes for Mar 28, 2017](https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-03-28.md)

Design some remaining 7.1 features

1. Fix pattern matching restriction with generics
2. Better best common type

Please discuss.

**Comment by DavidArno** [comment]
When looking at pattern matching restrictions with generics, please take [Is T expressions result in CS8121 error if sealed class, implementing a generic interface, is used for T](https://github.com/dotnet/roslyn/issues/16993), as this scenario has caused me to have to write more complex code on a number of occasions already.

**Comment by CyrusNajmabadi** [comment]
@DavidArno the PR to address that issue was merged here: https://github.com/dotnet/roslyn/pull/18784

Is there something additional you'd like done?  Thanks!

**Comment by DavidArno** [comment]
@CyrusNajmabadi

That's great news. Thanks. 

Re would I like anything else done, could v7.1 be released please, so I can make use of it? 😉
